File: app/src/main/java/com/example/project_note/Fragment/VisibleItemFragment.java
package com.example.project_note.Fragment;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentResultListener;
import androidx.fragment.app.FragmentTransaction;

import com.example.project_note.DataBase.Note;
import com.example.project_note.DataBase.NoteDatabase;
import com.example.project_note.R;

public class VisibleItemFragment extends Fragment {
    TextView mTxtTitle;
    TextView mTxtDetail;
    ImageButton mImageButtonBack;

    ImageButton mImageButtonEdit;
    private int id;
    Note note2;


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_visible_item,container,false);

        mTxtDetail = view.findViewById(R.id.txt_detail_visible);
        mTxtTitle = view.findViewById(R.id.txt_title_visible);
        mImageButtonBack = view.findViewById(R.id.imgbut_back);
        mImageButtonEdit = view.findViewById(R.id.imgbut_edit);
        mImageButtonBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                    if (getParentFragmentManager() != null) {
                        getParentFragmentManager().popBackStack();
                    }
            }
        });

        mImageButtonEdit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Note note = new Note(mTxtTitle.getText().toString(),mTxtDetail.getText().toString());
                openFragmentEdit(note,id);
            }
        });


        getParentFragmentManager().setFragmentResultListener("dataFromEdit", this, new FragmentResultListener() {
            @Override
            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle result) {
                Note note = (Note) result.getSerializable("note5");
                if(note != null) {
                    mTxtDetail.setText(note.getDetail());
                    mTxtTitle.setText(note.getTitle());
                    id = note.getId();
                }
            }
        });
        getParentFragmentManager().setFragmentResultListener("dataFromHome", this, new FragmentResultListener() {
            @Override
            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle result) {
                Note note = (Note) result.getSerializable("note");
                if(note != null) {
                    mTxtDetail.setText(note.getDetail());
                    mTxtTitle.setText(note.getTitle());
                    id = note.getId();
                }
            }
        });

        getParentFragmentManager().setFragmentResultListener("dataFromSearch", this, new FragmentResultListener() {
            @Override
            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle result) {
                Note note = (Note) result.getSerializable("note9");
                if(note != null) {
                    mTxtDetail.setText(note.getDetail());
                    mTxtTitle.setText(note.getTitle());
                    id = note.getId();
                }
            }
        });
        return view;
    }

    private void openFragmentEdit(Note note, int id) {

        EditNoteFragment editNoteFragment = new EditNoteFragment();

        Bundle bundle = new Bundle();
        bundle.putSerializable("note1",  note);
        bundle.putInt("KeyInt",id);
        getParentFragmentManager().setFragmentResult("dataFromVisible",bundle);

        FragmentManager fragmentManager = getParentFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.frag_layout,editNoteFragment );
        fragmentTransaction.addToBackStack(null); // Nếu bạn muốn thêm vào back stack
        fragmentTransaction.commit();
    }





    // xem sau
//    @Override
//    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
//        super.onActivityResult(requestCode, resultCode, data);
//        if (requestCode == 123 && resultCode == Activity.RESULT_OK && data != null) {
//            Bundle receivedBundle = data.getExtras();
//
//            if (receivedBundle != null) {
//                // Truy cập dữ liệu từ Bundle
//                  note2 = (Note) receivedBundle.getSerializable("note2");
//                 if(note2 != null){
//                     Toast.makeText(getContext(), note2.getTitle() + "", Toast.LENGTH_SHORT).show();
//                     Log.e("AAA","Đã vào đây rồi");
//                 }
//
//            }
//
//        }
//    }




}

File: app/src/main/java/com/example/project_note/fragment/VisibleItemFragment.kt
package com.example.project_note.fragment

import android.app.ProgressDialog
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import com.example.project_note.DataBase.Image
import com.example.project_note.DataBase.Note
import com.example.project_note.R
import com.example.project_note.RecycleView.IClickListener1
import com.example.project_note.RecycleView.ImageAdapter
import com.example.project_note.ViewModal.ViewModalImage
import com.example.project_note.ViewModal.ViewModalNote
import com.example.project_note.databinding.FragmentVisibleItemBinding

class VisibleItemFragment : Fragment() {

    lateinit var binding_Visible: FragmentVisibleItemBinding
    private val binding get() = binding_Visible!!

    private lateinit var mViewModalNote: ViewModalNote
    private lateinit var mViewModalImage: ViewModalImage

    private lateinit var mImageAdapter : ImageAdapter


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        Log.d("Visible","Sang Thang Visible rồi")
        binding_Visible = FragmentVisibleItemBinding.inflate(inflater, container,false)
        mViewModalNote = ViewModelProvider(requireActivity()).get(ViewModalNote::class.java)
        mViewModalImage = ViewModelProvider(requireActivity()).get(ViewModalImage::class.java)



        setupRecycleView()
        mViewModalNote.selectNote.observe(viewLifecycleOwner){

            binding_Visible.txtTitleVisible.setText(it?.title.toString())
            binding_Visible.txtDetailVisible.setText(it?.detail.toString())
            showdata(it)
        }
        setupButtons()
        mViewModalImage.setListImageDeleteNull()
        return binding.root
    }

    private fun setupRecycleView() {

        mImageAdapter = ImageAdapter(object : IClickListener1{
            override fun onClickImage(image: Image) {
                mViewModalImage.updateSelectImage(image)
                openImageFragment()
            }

        },requireContext())

        binding_Visible.rcyPhotos.adapter = mImageAdapter
    }

    private fun openImageFragment() {
        val bundle = Bundle()
        bundle.putString("keyvisible", "visible")
        parentFragmentManager.beginTransaction()
            .replace(R.id.frag_layout, ImageFragment::class.java,bundle)
            .addToBackStack(null)
            .commit()
    }

    private fun setupButtons() {
        binding_Visible.imgbutEdit.setOnClickListener {
            openFragmentEdit()
        }
        binding_Visible.imgbutBack.setOnClickListener {
            backPressed()
        }
    }

    private fun openFragmentEdit() {
        parentFragmentManager.beginTransaction()
            .replace(R.id.frag_layout, EditFragment())
            .addToBackStack(null)
            .commit()
    }

    private fun showdata(it: Note) {

        mViewModalImage.getListImage(it.idNote)
        mViewModalImage.mListImage.observe(viewLifecycleOwner){
            mImageAdapter.submitList(it)
        }
    }

    private fun backPressed(){
        val fragmentManager = requireActivity().supportFragmentManager
        if (fragmentManager.backStackEntryCount > 0) {
            fragmentManager.popBackStack()
        } else {
            requireActivity().finish()
        }
    }


}